internal/app/orchestrator/downstream.go

```diff
@@ -78,3 +80,17 @@ func (d *downstreamResponseMap) deleteAll(watchers map[*gcp.Request]bool) {
 		}
 	}
 }
+
+func (d *downstreamResponseMap) keys(m *mapper.Mapper) ([]string, error) {
```

Comment:
nit: maybe change this to `requestKeys` to not conflate with the keys for the map which are of `Request` type.